lms/djangoapps/grades/models.py

```diff
@@ -107,7 +107,8 @@ def hash_from_blockrecords(self, blocks):
             blocks = BlockRecordSet(blocks)
 
         try:
-            model = self.create_from_blockrecords(blocks)
+            with transaction.atomic():
+                model = self.create_from_blockrecords(blocks)
```

Comment:
@sanfordstudent your initial comment was correct. There were a lot of places in tests that had the query count lowered initially after the optimization, then reset after adding the atomic. There are still a few places remaining where the move to atomic yielded a net increase in queries.